src/main/java/sg/ncl/webssh/HttpHandshakeInterceptor.java

```diff
@@ -0,0 +1,32 @@
+package sg.ncl.webssh;
+
+import org.springframework.http.server.ServerHttpRequest;
+import org.springframework.http.server.ServerHttpResponse;
+import org.springframework.http.server.ServletServerHttpRequest;
+import org.springframework.web.socket.WebSocketHandler;
+import org.springframework.web.socket.server.HandshakeInterceptor;
+
+import javax.servlet.http.HttpSession;
+import java.util.Map;
+
+/**
+ * References:
+ * [1] http://www.devglan.com/spring-boot/spring-session-stomp-websocket
+ */
+
+public class HttpHandshakeInterceptor implements HandshakeInterceptor {
+
+    @Override
+    public boolean beforeHandshake(ServerHttpRequest request, ServerHttpResponse response,
+                                   WebSocketHandler handler, Map attributes) {
+        if (request instanceof ServletServerHttpRequest) {
+            ServletServerHttpRequest servletServerHttpRequest = (ServletServerHttpRequest) request;
+            HttpSession session = servletServerHttpRequest.getServletRequest().getSession();
+            attributes.put("sessionId", session.getId());
+        }
+        return true;
+    }
+
+    public void afterHandshake(ServerHttpRequest request, ServerHttpResponse response,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)